### PR TITLE
chore: deprecate pip-compile-multi in favor or uv

### PR DESCRIPTION
**Pull request by mistercrunch**
pip-compile-multi hasn't been working super well for us. It has a few bugs, is pretty slow, and turns out `uv` supports our use case much better and at blazing speeds.

### CODE REVIEW

File: scripts/uv-pip-compile.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+set -e
+
+ADDITIONAL_ARGS="$@"
+
+# Generate the requirements/base.txt file
+uv pip compile pyproject.toml -c requirements/base.in -o requirements/base.txt $ADDITIONAL_ARGS
```
Comment (EugeneTorap) [inline scripts/uv-pip-compile.sh:25]: You can provide `--python-version=3.10` and `--python-platform=linux` params in order to ensure that the output file is the same if the user uses a different Python version (like Py3.9 or Py3.11) or OS (like MacOS or Windows)

Comment (mistercrunch) [reply scripts/uv-pip-compile.sh:25]: I think that ruff/uv both take into account this [directive in pyproject.tom](https://github.com/apache/superset/blob/master/pyproject.toml#L27). Also we still support `3.9` as our build system does some CI steps for previous/current/next where previous is 3.9 at the moment.

File: scripts/uv-pip-compile.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
```
[the remark this answers is not in the document]
Comment (rusackas) [reply scripts/uv-pip-compile.sh:25]: @mistercrunch is this accurate? ¯\\\_(ツ)_/¯

File: scripts/uv-pip-compile.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+set -e
+
+ADDITIONAL_ARGS="$@"
```
[the remark this answers is not in the document]
Comment (mistercrunch) [reply scripts/uv-pip-compile.sh:22]: who in the world validates that commands exist before using them?

File: scripts/uv-pip-compile.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (mistercrunch) [reply scripts/uv-pip-compile.sh:25]: `requirements.in` really is used as a set of constraints here, the name might be confusing though. I could rename to `base-constraints.txt` for clarity

[the remark this answers is not in the document]
Comment (mistercrunch) [reply scripts/uv-pip-compile.sh:25]: I could probably also merge the constraints in `pyproject.toml` so it's all in one place

**Review by eschutho — APPROVED** [review]

**Comment by andy-clapson** [comment]
Hey @mistercrunch very nice to see! It's crazy fast, hey?

**Pull request merged**